### PR TITLE
Workaround error in Overlay when site has no URLs

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

When creating a site it is possible to not specify URLs. In this case, we send a plugin setting entry in the HTTP request as `['name' => 'urls']` w/ no `value` key. Currently this is ignored by Matomo, and the validation that happens does not occur. Which allows sites to have no URLs.

If this happens and Overlay is loaded, the JS will error when trying to find the host of the site URLs.

This is fixed in this PR w/ tests. The issue in Overlay is also worked around in case there are users who created sites w/ no URLs.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/SitesManager/API.php [hunk 1]
```diff
@@ -598,7 +598,7 @@ public function addSite($siteName,
         }
 
         $coreProperties = array();
-        $coreProperties = $this->setSettingValue('urls', $urls, $coreProperties, $settingValues);
+        $coreProperties = $this->setSettingValue('urls', $urls, $coreProperties, $settingValues, $isRequired = true);
```
Comment (sgiehl) [inline plugins/SitesManager/API.php]: This makes it impossible to create any new roll ups, as for roll ups you can't define any urls.

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: plugins/SitesManager/API.php (hunk 1 above)
Comment (diosmosis) [reply plugins/SitesManager/API.php]: Good point. I guess I can make it required if type == 'website'. (cc @tsteur)

**Comment by tsteur** [comment]
@diosmosis this would be breaking API (even though in the code it should have already worked but from a user perspective things are potentially still breaking as it probably wasn't working for a while) and not sure if want to require URLs now or maybe wait to Matomo 5. Is there an issue for this PR?

**Comment by diosmosis** [comment]
@tsteur this is to fix the root cause of L3-65

**Comment by diosmosis** [comment]
@tsteur updated. Warning looks like:

image

**Comment by tsteur** [comment]
👍  be great to have an issue to fix the addSite/updateSite API regarding requiring URLs as part of Matomo 5

**Comment by diosmosis** [comment]
Created https://github.com/matomo-org/matomo/issues/17459
